pinot-core/src/main/java/org/apache/pinot/core/metadata/DefaultMetadataExtractor.java

```diff
@@ -35,7 +36,12 @@ public SegmentMetadata extractMetadata(File tarredSegmentFile, File unzippedSegm
       throws Exception {
     // NOTE: While there is TarGzCompressionUtils.untarOneFile(), we use untar() here to unpack all files in the segment
     //       in order to ensure the segment is not corrupted.
-    File indexDir = TarCompressionUtils.untar(tarredSegmentFile, unzippedSegmentDir).get(0);
-    return new SegmentMetadataImpl(indexDir);
+    List<File> untarredFiles = TarCompressionUtils.untar(tarredSegmentFile, unzippedSegmentDir);
+    File indexDir = untarredFiles.get(0);
+    if (indexDir.isDirectory()) {
+      return new SegmentMetadataImpl(indexDir);
+    } else {
+      return new SegmentMetadataImpl(untarredFiles);
```

Comment:
An easier way to handle this is to create a directory and move the files into the directory. This way we don't need to change other classes